### ISSUE TITLE
moving_mean and moving_variance are not updated in the batch_normalization layers

### ISSUE THREAD

**Issue by maxjeblick** [ImagingLab/Colorizing-with-GANs#5]
To my understanding, the moving mean and moving average values in the batch normalization layers are not updated during training. According to the tensorflow documentation, these values are not included  in the
tf.GraphKeys.TRAINABLE_VARIABLES, but are placed in tf.GraphKeys.UPDATE_OPS.

You can also inspect the pretrained weights using
`from tensorflow.python.tools.inspect_checkpoint import print_tensors_in_checkpoint_file
tf.print_tensors_in_checkpoint_file('/root/checkpoints/CGAN_places365', all_tensors=True, tensor_name='')`
to see that the moving_mean and moving_variance values are equal to their default value.

**Comment by knazeri** [comment]
@maxjeblick  Batch-Normalization with statistics of the test batch, rather than aggregated (moving average) statistics of the training can arguably produce better quality images in generative models. This is why we don't include the train batch statistics of the normalization layers. Same idea is also mentioned in the [Pix2Pix paper](https://arxiv.org/abs/1611.07004).

The layer could be replaced with the newly proposed [Instance-Normalization ](https://arxiv.org/abs/1607.08022) that applies normalization at both test and train time using image statistics.

**Comment by maxjeblick** [comment]
Thank you for the kind clarification!
